### PR TITLE
ReplyM3: Adaptive and Fold state setup

### PR DESCRIPTION
**Pull request by Gurupreet**
- Setup size similar to [jetnews](https://github.com/android/compose-samples/blob/main/JetNews/app/src/main/java/com/example/jetnews/utils/WindowSize.kt) for adaptive sizing
- Setup Fold state similar to [JetCaster](https://github.com/android/compose-samples/blob/main/Jetcaster/app/src/main/java/com/example/jetcaster/ui/MainActivity.kt)
- Use proper navigation and content depending on the Screen Size and Fold state.
Screen Shot 2022-03-15 at 13 17 18

### CODE REVIEW

File: Reply/app/build.gradle
Comment (ricknout) [inline Reply/app/build.gradle:55]: `alpha07` is available :)

File: Reply/app/src/main/java/com/example/reply/MainActivity.kt
Comment (ricknout) [inline Reply/app/src/main/java/com/example/reply/MainActivity.kt:41]: Does this work even if the app is launched while in a different posture?

File: Reply/app/src/main/java/com/example/reply/MainActivity.kt
Comment (ricknout) [inline Reply/app/src/main/java/com/example/reply/MainActivity.kt:23]: Is this fine to be in `MainActivity`? Do you plan to use a `ViewModel` or something later on?

File: Reply/app/src/main/java/com/example/reply/MainActivity.kt
Comment (ricknout) [inline Reply/app/src/main/java/com/example/reply/MainActivity.kt:12]: Usually I think we try to avoid * imports if possible

File: Reply/app/src/main/java/com/example/reply/ui/WindowStateUtils.kt
Comment (ricknout) [inline Reply/app/src/main/java/com/example/reply/ui/WindowStateUtils.kt:43]: nit: some enum classes in this file are UPPERCASE whereas others are CamelCase. Choose one convention for all of them

File: Reply/app/src/main/java/com/example/reply/ui/WindowStateUtils.kt
Comment (ricknout) [inline Reply/app/src/main/java/com/example/reply/ui/WindowStateUtils.kt:41]: Is there a corresponding m3.material.io link?

Comment (Gurupreet) [reply Reply/app/src/main/java/com/example/reply/ui/WindowStateUtils.kt:41]: Hmm not sure let me check on that.

File: Reply/app/src/main/java/com/example/reply/MainActivity.kt
Comment (alexvanyo) [inline Reply/app/src/main/java/com/example/reply/MainActivity.kt:23]: Small nit: I think it'd be clearer if `WindowInfoTracker` was kept at the front here, the direct import is a bit confusing:
```suggestion
        val devicePosture = WindowInfoTracker.getOrCreate(this).windowLayoutInfo(this)
```

Comment (alexvanyo) [reply Reply/app/src/main/java/com/example/reply/MainActivity.kt:23]: My understanding is this has to be in `MainActivity`, since the layout info requires an `Activity` to query.

File: Reply/app/src/main/java/com/example/reply/ui/ReplyApp.kt
Comment (alexvanyo) [inline Reply/app/src/main/java/com/example/reply/ui/ReplyApp.kt:47]: Optional Kotlin trick:

You can separate the declaration from the initialization, and Kotlin will be smart enough to verify that in all cases it was initialized once and exactly once. That might be clearer than the `Pair` approach:

```suggestion
    val navigationType: ReplyNavigationType
    val contentType: ReplyContentType
    when (windowSize) {
        WindowSize.Compact -> {
            replyNavigationType = ReplyNavigationType.BOTTOM_NAVIGATION
            replyContentType = ReplyContentType.LIST_ONLY
        }
        WindowSize.Medium -> {
            replyNavigationType = ReplyNavigationType.NAVIGATION_RAIL
            replyContentType = if (foldingDevicePosture != DevicePosture.NormalPosture) {
                ReplyContentType.LIST_AND_DETAIL
            } else {
                ReplyContentType.LIST_ONLY
            }
        }
        WindowSize.Expanded -> {
            replyNavigationType = if (foldingDevicePosture is DevicePosture.BookPosture) {
               ReplyNavigationType.NAVIGATION_RAIL
            } else {
               ReplyNavigationType.FIXED_NAVIGATION_DRAWER
            }
            replyContentType = ReplyContentType.LIST_AND_DETAIL
        }
    }
```

File: Reply/app/src/main/java/com/example/reply/ui/ReplyApp.kt
Comment (Gurupreet) [reply Reply/app/src/main/java/com/example/reply/ui/ReplyApp.kt:54]: Hey yes you are right when Bottom Navigation is active it will not show navigation drawer. With Navigation Rail it will be shown so for simplicity I will keep Drawer as main top wrapper and shown or not will be decided on dynamic states of fold and size.

File: Reply/app/src/main/java/com/example/reply/ui/ReplyApp.kt
Comment (Gurupreet) [reply Reply/app/src/main/java/com/example/reply/ui/ReplyApp.kt]: Added :)

**Review by ricknout — APPROVED** [review]
Happy that my comments were addressed but would wait for Alex's approval

File: Reply/app/src/main/java/com/example/reply/ui/ReplyApp.kt
Comment (alexvanyo) [reply Reply/app/src/main/java/com/example/reply/ui/ReplyApp.kt:57]: Ah gotcha, so this will be handled in upcoming code to managed the drawer state

**Review by alexvanyo — APPROVED** [review]
Looks good to me, super interested to see how this evolves!

**Pull request merged**
